pkg/retriever/reportretriever.go

```diff
@@ -205,145 +201,33 @@ func (r *Retriever) CallInsights(req *http.Request, cluster types.ManagedCluster
-	// the task https://github.com/RedHatInsights/insights-results-smart-proxy/issues/450
-	// is completed
-	userAgent := "insights-operator/v1.0.0+b653953-b653953ed174001d5aca50b3515f1fa6f6b28728 cluster/" + clusterId
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("User-Agent", userAgent)
-	req.Header.Set("Authorization", "Bearer "+r.Token)
-	return req, nil
-}
 
-// Makes HTTP call to get contents
-func (r *Retriever) CallContents(req *http.Request) (types.ContentsResponse, error) {
-	glog.Infof("Making GET call for  Contents .")
-	var responseBody types.ContentsResponse
-	res, err := r.Client.Do(req)
-	if err != nil {
-		glog.Warningf("Error sending HttpRequest for contents %v", err)
-		return types.ContentsResponse{}, err
-	}
-	if res.StatusCode != 200 {
-		glog.Warningf("Unsucessful response during contents GET - response code %d", res.StatusCode)
-		return types.ContentsResponse{}, e.New("No Success HTTP Response code ")
-	}
-	defer res.Body.Close()
-	data, _ := ioutil.ReadAll(res.Body)
-	// unmarshal response data into the ResponseBody struct
-	unmarshalError := json.Unmarshal(data, &responseBody)
-	if unmarshalError != nil {
-		glog.Errorf("Error unmarshalling ResponseBody %v", unmarshalError)
-		return types.ContentsResponse{}, unmarshalError
-	}
-	return responseBody, err
-}
-
-// Function to make a GET HTTP call to get all the contents for reports
-func (r *Retriever) retrieveCCXContent(clusterId string) int {
-	req, err := r.GetContentRequest(context.TODO(), clusterId)
-	if err != nil {
-		glog.Warningf("Error creating HttpRequest with endpoint %s, %v", r.ContentUrl, err)
-		return -1
-	}
-	contents, err := r.CallContents(req)
-	if err != nil {
-		glog.Warningf("Error calling for contents %s, %v", clusterId, err)
-		return -1
-	}
-	r.createContents(contents)
-	return len(contentsMap)
-}
-
-func (r *Retriever) InitializeContents(hubId string) int {
-	return r.retrieveCCXContent(hubId)
-}
-
-// Populate json response from /contents call onto a Map to quick lookup
-func (r *Retriever) createContents(responseBody types.ContentsResponse) {
-	glog.Infof("Creating Contents from json ")
-	contentsMap = make(map[string]map[string]interface{})
-
-	for content := range responseBody.Content {
-		for errorName, errorVal := range responseBody.Content[content].Error_keys {
-			errorMap := make(map[string]interface{})
-			errorMap["summary"] = responseBody.Content[content].Summary
-			errorMap["reason"] = responseBody.Content[content].Reason
-			errorMap["resolution"] = responseBody.Content[content].Resolution
-			errorVals := errorVal.(map[string]interface{})
-			for key, val := range errorVals {
-				if key == "metadata" {
-					errorMap = r.getErrorKey(val, errorMap)
-				} else {
-					errorMap[key] = val
-				}
-			}
-			lock.Lock()
-			contentsMap[errorName] = errorMap
-			lock.Unlock()
+			return types.ProcessorData{
+				ClusterInfo: cluster,
+				Reports: reports,
+				Contents: contentsMap,
```

Comment:
@xrajesh Im not sure how else to get the contents to the processor, but I dont really like the idea of passing the entire contents object every time. Suggestions?

Reply:
@zlayne the contents Map is going to be static , and periodically refreshed . We dont need to pass every time .  We can export that map, and we could access in reportprocessor.go